lib/casbin/config/config.rb

```diff
@@ -2,20 +2,106 @@
 module Casbin
   module Config
     class Config
+      # DEFAULT_SECTION specifies the name of a section if no name provided
+      DEFAULT_SECTION = 'default'
+      # DEFAULT_COMMENT defines what character(s) indicate a comment `#`
+      DEFAULT_COMMENT = '#'
+      # DEFAULT_COMMENT_SEM defines what alternate character(s) indicate a comment `;`
+      DEFAULT_COMMENT_SEM = ';'
+      # DEFAULT_MULTI_LINE_SEPARATOR defines what character indicates a multi-line content
+      DEFAULT_MULTI_LINE_SEPARATOR = '\\'
+
+      attr_reader :data
+
+      def initialize
+        @data = {}
+      end
+
       def self.new_config(conf_name)
-        # stub method
-        puts "call new_config with conf_name = #{conf_name}"
+        config = Casbin::Config::Config.new
+        config.parse(conf_name)
+        config
       end
 
       def set(key, value)
-        # stub method
-        puts "call get with key = #{key} value = #{value}"
+        raise "key is empty" if key.size.zero?
+
+        keys = key.downcase.split('::')
+        if keys.size >= 2
+          section = keys[0]
+          option = keys[1]
+        else
+          section = ""
+          option = keys[0]
+        end
+
+        add_config(section, option, value)
       end
 
       # section.key or key
       def get(key)
-        # stub method
-        puts "call get with key = #{key}"
+        keys = key.downcase.split('::')
```

Comment:
It will raise undefined method too if key is nil. Try convert to string  (`to_s`) or check Nill class.